src/main/scala/mesosphere/marathon/core/launchqueue/impl/ReviveOffersState.scala

```diff
@@ -0,0 +1,99 @@
+package mesosphere.marathon
+package core.launchqueue.impl
+
+import com.typesafe.scalalogging.StrictLogging
+import mesosphere.marathon.core.instance.update.InstancesSnapshot
+import mesosphere.marathon.core.instance.{Goal, Instance}
+import mesosphere.marathon.state.RunSpecConfigRef
+
+/**
+  * Holds the current state and defines the revive logic.
+  *
+  * @param scheduledInstances All scheduled instance requiring offers and their run spec ref.
+  * @param terminalReservations Ids of terminal resident instance with [[Goal.Decommissioned]].
+  * @param activeDelays Delays for run specs.
+  */
+case class ReviveOffersState(
+    scheduledInstances: Map[Instance.Id, RunSpecConfigRef],
+    terminalReservations: Set[Instance.Id],
+    activeDelays: Set[RunSpecConfigRef]) extends StrictLogging {
+
+  // At least one instance with reservations was force expunged before the reservation was destroyed.
+  // This field should not be copied.
+  private var forceExpungedResidentInstance: Boolean = false
```

Comment:
This is an issue. The field is `true` once and might be dropped during debouncing.